src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java19/ExpectedJava19PatternsTrickyWhenUsage.txt

```diff
@@ -141,9 +142,10 @@ COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
 |       |       |   |       |   |           |   |--TYPE -> TYPE [20:17]
 |       |       |   |       |   |           |   |   `--IDENT -> Integer [20:17]
 |       |       |   |       |   |           |   `--IDENT -> when [20:25]
-|       |       |   |       |   |           `--GE -> >= [20:40]
-|       |       |   |       |   |               |--IDENT -> when [20:35]
-|       |       |   |       |   |               `--NUM_INT -> 0 [20:43]
+|       |       |   |       |   |           `--EXPR -> EXPR [20:40]
+|       |       |   |       |   |               `--GE -> >= [20:40]
+|       |       |   |       |   |                   |--IDENT -> when [20:35]
```

Comment:
I see a concern and will put it in a new issue.

Reply:
https://github.com/checkstyle/checkstyle/blob/d951e083e6f631937374678adfbeafc67fe45424/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java19/InputJava19PatternsTrickyWhenUsage.java#L18-L22